plugins/SegmentEditor/API.php

```diff
@@ -351,6 +368,29 @@ public function getAll($idSite = false)
     }
 
     /**
+     * Returns stored segments that are set to be archived during cron archiving.
+     *
+     * @param bool|false $idSite
+     * @return array
+     */
+    public function getSegmentsToAutoArchive($idSite = false)
+    {
+        if (!empty($idSite)) {
+            Piwik::checkUserHasAdminAccess($idSite);
+        } else {
+            Piwik::checkUserHasSuperUserAccess();
+        }
+
+        $cacheKey = 'SegmentEditor.getSegmentsToAutoArchive_' . (!empty($idSite) ? $idSite : 'enabled_all');
```

Comment:
should we cast idSite to `int` just for safety and to prevent possible security issues? Eg if there was another key starting with the same cache key one could maybe read another cache key and it could then be exposed eg via an error message or so (if the contained data has different structure to expected one). It's a rare case and possibly `checkUserHasAdminAccess()` would fail if it is not a valid integer but not sure. Also `checkUserHasAdminAccess()` accepts arrays so it might have to be an `implode(',', $idSite)` if `$idSite` is an array.

I checked and `checkUserHasAdminAccess` should currently fail if it's not a number but would be maybe more secure though to cast to int or to do an implode on top if it is an array